cgotorch/cgotorch.h

```diff
@@ -27,6 +28,32 @@ void Step(Optimizer opt);
 void AddParameters(Optimizer opt, Tensor *tensors, int length);
 void Optimizer_Close(Optimizer opt);
 
+// transform APIs
+typedef void *Transform;
+Transform Normalize(double mean, double stddev);
+Transform Stack();
+
+// dataset APIs
+typedef void *Dataset;
+Dataset MNIST(const char *data_root);
+
+// Add transform on dataset
+void AddNormalize(Dataset dataset, Transform transform);
+void AddStack(Dataset dataset, Transform transform);
+
+typedef void *Iterator;
+typedef void *DataLoader;
+
+typedef struct Data{
+  Tensor Data;
+  Tensor Target;
+} Data;
```

Comment:
It seems that `Data` could be an array of Tensors.  In the [DCGAN example](https://github.com/pytorch/examples/blob/master/dcgan/main.py#L225), the data loader returns an array of tensors. 

And, another question, how do users define their own dataset? Maybe the user-defined dataset has three fields.

Reply:
> It seems that Data could be an array of Tensors.

That may be confused to users,  PyTorch Python/C++ API also provides the `data` and `target` member:

c.f. [Python example](https://github.com/pytorch/examples/blob/8ec17a42412c3cb76bd8e2392e8a4fea5718ebb0/mnist/main.py#L39)
``` python
    for batch_idx, (data, target) in enumerate(train_loader):
        data, target = data.to(device), target.to(device)
        optimizer.zero_grad()
        output = model(data)
```
c.f. [C++ example](https://github.com/pytorch/examples/blob/master/cpp/mnist/mnist.cpp#L65)
``` c++
  for (auto& batch : data_loader) {
    auto data = batch.data.to(device), targets = batch.target.to(device);
    optimizer.zero_grad();
    auto output = model.forward(data);
```